### PR TITLE
feat(core): add external configuration for lag calculation window size

### PR DESCRIPTION
**Pull request by bluestreak01**
Ability to set size of the window, which is used by O3 automatic lag algo to maintain correct lag size. Before this lag window was hardcoded to have 4 entries, which may be insufficient for some ingestion data patterns.

The new option can be set via:

```
cairo.o3.lag.calculation.windows.size=128
```

### CODE REVIEW

**Review by ideoma — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

File: core/src/main/java/io/questdb/PropServerConfiguration.java [hunk 1]
```diff
@@ -897,6 +898,7 @@ public PropServerConfiguration(
             this.o3PartitionQueueCapacity = getQueueCapacity(properties, env, PropertyKey.CAIRO_O3_PARTITION_QUEUE_CAPACITY, 128);
             this.o3OpenColumnQueueCapacity = getQueueCapacity(properties, env, PropertyKey.CAIRO_O3_OPEN_COLUMN_QUEUE_CAPACITY, 128);
             this.o3CopyQueueCapacity = getQueueCapacity(properties, env, PropertyKey.CAIRO_O3_COPY_QUEUE_CAPACITY, 128);
+            this.o3LagCalculationWindowsSize = getIntSize(properties, env, PropertyKey.CAIRO_O3_LAG_CALCULATION_WINDOW_SIZE, 4);
```
Comment (marregui) [inline core/src/main/java/io/questdb/PropServerConfiguration.java:901]: in server conf:

```
cairo.o3.lag.calculation.windows.size=120
```

just making sure it is intentional.

**Review by marregui — APPROVED** [review]
LGTM

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

File: core/src/main/java/io/questdb/PropServerConfiguration.java (hunk 1 above)
Comment (ideoma) [reply core/src/main/java/io/questdb/PropServerConfiguration.java:901]: This is test config

**Pull request merged**